### PR TITLE
feat(api): add `defaultWindowIcon` to `app` module

### LINKED ISSUE #9957: [feat] Front-end implementation to obtain default icon API.
### Describe the problem

The front-end uses JavaScript to obtain the default icon API.

### Describe the solution you'd like

**app.defaultWindowIcon()**

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
closes #9957

### CODE REVIEW

File: .changes/api-app-defaultWindowIcon.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"tauri": "patch:feat"
+"@tauri-apps/api": "patch:api"
+---
```
Comment (amrbashir) [inline .changes/api-app-defaultWindowIcon.md]:
```suggestion
"tauri": "patch:feat"
"@tauri-apps/api": "patch:feat"
---
```

File: core/tauri/permissions/app/autogenerated/reference.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 |`deny-app-hide`|Denies the app_hide command without any pre-configured scope.|
 |`allow-app-show`|Enables the app_show command without any pre-configured scope.|
 |`deny-app-show`|Denies the app_show command without any pre-configured scope.|
+|`allow-default-window-icon`|Enables the default_window_icon command without any pre-configured scope.|
+|`deny-default-window-icon`|Denies the default_window_icon command without any pre-configured scope.|
+|`allow-get-default-window-icon`|Enables the get_default_window_icon command without any pre-configured scope.|
+|`deny-get-default-window-icon`|Denies the get_default_window_icon command without any pre-configured scope.|
```
Comment (FabianLars) [inline core/tauri/permissions/app/autogenerated/reference.md]: Is this a leftover or is our generator broken?

Comment (amrbashir) [reply core/tauri/permissions/app/autogenerated/reference.md]: this is a leftover from a previous iteration, I will regenerate

Comment (amrbashir) [inline core/tauri/permissions/app/autogenerated/reference.md]:
```suggestion
```

**Review by FabianLars — APPROVED** [review]

**Pull request merged**